zerver/views/realm_logo.py

```diff
@@ -2,52 +2,77 @@
     if user_profile.realm.plan_type == Realm.LIMITED:
         return json_error(_("Feature unavailable on your current plan."))
 
     if len(request.FILES) != 1:
         return json_error(_("You must upload exactly one logo."))
-
     logo_file = list(request.FILES.values())[0]
     if ((settings.MAX_LOGO_FILE_SIZE * 1024 * 1024) < logo_file.size):
         return json_error(_("Uploaded file is larger than the allowed limit of %s MB") % (
             settings.MAX_LOGO_FILE_SIZE))
-    upload_logo_image(logo_file, user_profile)
-    do_change_logo_source(user_profile.realm, user_profile.realm.LOGO_UPLOADED)
-    logo_url = realm_logo_url(user_profile.realm)
-
-    json_result = dict(
-        logo_url=logo_url
-    )
+    if night:
+        upload_logo_image(logo_file, user_profile, night)
+        do_change_logo_source(user_profile.realm, user_profile.realm.NIGHT_LOGO_UPLOADED, night)
+        logo_url = realm_night_logo_url(user_profile.realm)
+        json_result = dict(
+            night_logo_url=logo_url
+        )
+    else:
+        upload_logo_image(logo_file, user_profile, night)
+        do_change_logo_source(user_profile.realm, user_profile.realm.LOGO_UPLOADED, night)
+        logo_url = realm_logo_url(user_profile.realm)
+        json_result = dict(
+            logo_url=logo_url
+        )
     return json_success(json_result)
 
-
 @require_realm_admin
-def delete_logo_backend(request: HttpRequest, user_profile: UserProfile) -> HttpResponse:
+@has_request_variables
+def delete_logo_backend(request: HttpRequest, user_profile: UserProfile,
+                        night: bool=REQ(validator=check_bool)) -> HttpResponse:
     # We don't actually delete the logo because it might still
     # be needed if the URL was cached and it is rewrited
     # in any case after next update.
-    do_change_logo_source(user_profile.realm, user_profile.realm.LOGO_DEFAULT)
-    default_url = realm_logo_url(user_profile.realm)
-    json_result = dict(
-        logo_url=default_url
-    )
+    if night:
+        do_change_logo_source(user_profile.realm, user_profile.realm.NIGHT_LOGO_DEFAULT, night)
+        default_url = realm_night_logo_url(user_profile.realm)
+        json_result = dict(
+            night_logo_url=default_url
+        )
+    else:
+        do_change_logo_source(user_profile.realm, user_profile.realm.LOGO_DEFAULT, night)
+        default_url = realm_logo_url(user_profile.realm)
+        json_result = dict(
+            logo_url=default_url
+        )
     return json_success(json_result)
 
-
-def get_logo_backend(request: HttpRequest, user_profile: UserProfile) -> HttpResponse:
-    url = realm_logo_url(user_profile.realm)
+@has_request_variables
+def get_logo_backend(request: HttpRequest, user_profile: UserProfile,
+                     night: bool=REQ(validator=check_bool)) -> HttpResponse:
+    if night:
+        url = realm_night_logo_url(user_profile.realm)
+    else:
+        url = realm_logo_url(user_profile.realm)
```

Comment:
I think if we make `realm_logo_url` take a `night` argument, we can get rid of the conditional here and just do `realm_logo_url(user_profile.realm, night)`.

That is a helpful refactor, because this pattern is repeated in a few place in this file.